### PR TITLE
tests: introduce e2e suite

### PR DESCRIPTION
**Pull request by pires**
This PR introduces an e2e (end-to-end) test suite containing basic tests that validate the following behavior:
- basic pod lifecycle management, including create and delete.
- pod metrics provider.

Since the e2e suite runs out-of-process, and in order to prove pods are indeed created and deleted, I needed to increase the _observability_ of the `mock` provider. I did this by implementing the `PodMetricsProvider`, which then allows us to know which pods are currently known to the mock provider by querying `<vk-host>:10255/stats/summary`.

**Notes to reviewers**:
- This implementation returns dummy CPU and RAM stats for pods currently known (as well as for their containers). 
- Instructions on how to run the test suite have been added to `README.md`.
- For CI, a new `e2e` job has been introduced to CircleCI.
Said job launches a Minikube cluster and deploys the VK (with the `mock` provider) as a Kubernetes pod by leveraging on [`skaffold`](https://github.com/GoogleContainerTools/skaffold).
Then, it runs the e2e test suite.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -108,6 +108,37 @@ format: $(GOPATH)/bin/goimports
 	$Q find . -iname \*.go | grep -v \
         -e "^$$" $(addprefix -e ,$(IGNORED_PACKAGES)) | xargs goimports -w
 
+# skaffold.run deploys the virtual-kubelet to the Kubernetes cluster targeted by the current kubeconfig using skaffold.
+.PHONY: skaffold.run
+skaffold.run: MODE ?= dev
```
Comment (cpuguy83) [inline Makefile]: Maybe a `skaffold.clean` as well?

File: test/e2e/basic_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	}()
+
+	// Create a pod with prefix "nginx-1-" having a single container.
+	pod1, err := f.CreatePod(f.CreateDummyPodObjectWithPrefix("nginx-1-", "bar"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Delete the "nginx-1-Y" pod after the test finishes.
+	defer func() {
+		if err := f.DeletePod(pod0.Namespace, pod0.Name); err != nil && !apierrors.IsNotFound(err) {
```
Comment (robbiezhang) [inline test/e2e/basic_test.go]: delete pod1, too?

Comment (pires) [reply test/e2e/basic_test.go]: Ah, copy'n'paste engineering 🤦‍♂️ thanks @robbiezhang will fix.

File: Makefile (hunk 1 above)
Comment (pires) [reply Makefile]: I agree we need a goal for cleaning up. `skaffold` already provides a `MODE=delete` so let's use that. I'm renaming the current goal from `skaffold.run` to `skaffold` and changing docs to document all `MODE` and `PROFILE`.

File: test/e2e/basic_test.go (hunk 2 above)
Comment (pires) [reply test/e2e/basic_test.go]: Done.

File: Makefile (hunk 1 above)
Comment (pires) [reply Makefile]: `make skaffold MODE=delete` is now documented.

**Pull request merged**